legend-pure-runtime/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/type/_class/ClassImplProcessor.java

```diff
@@ -1081,6 +1061,50 @@ public static String validate(boolean stateAndDeep, CoreInstance _class, String
                                 "        }\n" : "") +
                         "        return " + (stateAndDeep ? "this" : "null") + ";\n" +
                         "    }\n";
+     return validate + validateItems.toString();
+
+    }
+
+    private static String validateItem(boolean stateAndDeep, CoreInstance constraint, CoreInstance _class, ProcessorContext processorContext, String extraValues, String constraintIndex)
```

Comment:
It's better to take the constraint index as an int.
```suggestion
    private static String validateItem(boolean stateAndDeep, CoreInstance constraint, CoreInstance _class, ProcessorContext processorContext, String extraValues, int constraintIndex)
```